gdrive_sync/api.py

```diff
@@ -134,11 +134,16 @@ def import_recent_videos(last_dt=None):
         api_call=DRIVE_API_FILES
     )
     query = "(mimeType contains 'video/' and not trashed)"
+    fields = "nextPageToken, files(id, name, md5Checksum, mimeType, createdTime, modifiedTime, webContentLink, trashed, parents)"
     last_checked = last_dt or file_token_obj.last_dt
     if last_checked:
         dt_str = last_checked.strftime("%Y-%m-%dT%H:%M:%S.%f")
         query += f" and (modifiedTime > '{dt_str}' or createdTime > '{dt_str}')"
-    videos = get_file_list(query=query)
+
+    if settings.DRIVE_VIDEO_UPLOADS_PARENT_FOLDER_ID:
+        query += f" and ('{settings.DRIVE_VIDEO_UPLOADS_PARENT_FOLDER_ID}' in parents)"
```

Comment:
Unfortunately the `parents` field only returns the immediate parent folder, so as a result no videos will be returned if this is included in the query.  You might be able to make use of the `get_parent_tree` function to check if any of the returned parents have an id == `settings.DRIVE_VIDEO_UPLOADS_PARENT_FOLDER_ID` (if it is set to anything) and if not, skip processing it.

Reply:
That's a good catch